coalib/results/Diff.py

```diff
@@ -12,24 +12,34 @@ class Diff:
     A Diff result represents a difference for one file.
     """
 
-    def __init__(self, file_list, filename, rename=False, delete=False):
+    def __init__(self, file_list, filename=None, rename=False, delete=False):
         """
         Creates an empty diff for the given file.
 
+        Test to see if the deprecation warning prints properly.
+
+        >>> diff = Diff([])
```

Comment:
I used a doctest here, as I think it's the cleanest way to catch and test the print.